pkgs/development/python-modules/publicsuffix2/default.nix

```diff
@@ -1,13 +1,15 @@
 { lib, buildPythonPackage, fetchFromGitHub }:
-
-buildPythonPackage rec {
+let
+  tagVersion = "2.2019-12-21";
+in
+buildPythonPackage {
   pname = "publicsuffix2";
-  version = "2.2019-12-21";
+  version = lib.replaceStrings ["-"] [""] tagVersion;
 
   src = fetchFromGitHub {
     owner = "nexB";
     repo = "python-publicsuffix2";
-    rev = "release-${version}";
+    rev = "release-${tagVersion}";
```

Comment:
> `update-python-libraries` currently looks to GitHub releases, not tags, for updates, and python-publicsuffix2 [has no releases](https://github.com/nexB/python-publicsuffix2/releases).

I am using nix-update which should work. I also doubt that update-python-libraries requires a release, tags should be fine.

> but wouldn't such a rule be overridden by a canonical version number in a registry like PyPI?

Well, depends what the library itself thinks about its version number. Generally PyPi is usually not the best source for anything in my experience and needs to be taken with a big grain of salt.

> Isn't the fact that we fetch from GitHub here an implementation detail, and the version number not something that should change if we only switched to using `fetchPypi`?

Yes, but upstreams seems to not think like that.

> Again, this library is the only Python library to do this in all of Nixpkgs, and we are in the overwhelming minority on Repology on how this library is versioned (in fact, this causes the r-ryantm auto-updating infrastructure to repeatedly check on this package because it thinks our version is outdated, which is how I ended up looking at this in the first place).

This seems to be more like it is missing a rule in repology.